remote-trait-object/src/service/id.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use super::MethodId;
+use linkme::distributed_slice;
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, HashSet};
+
+pub const ID_ORDERING: std::sync::atomic::Ordering = std::sync::atomic::Ordering::SeqCst;
+pub type MethodIdAtomic = std::sync::atomic::AtomicU32;
+
+// linkme crate smartly collects all the registrations generated by the proc-macro
+// into a sinlge array in the link time.
+// Note that too long linkme-related variable name would cause serious compiler error in MacOS
+// So we deliberately make it have a short name
+
+// Id of methods in services.
+// Note that here the two strings mean (trait name, method name)
+// Also you can skip calling this, then the method id will be set up for default value
+// decided by the order of declaration.
+type MethodIdentifierSetter = fn(id: MethodId);
+#[distributed_slice]
+pub static MID_REG: [(&'static str, &'static str, MethodIdentifierSetter)] = [..];
```

Comment:
Using a global variable in the Foundry context seems ok.
However, if we want to publish "remote-trait-object" as a general library, this can be a hidden trap for users.
They can meet a unexpected erorr that uses the same service name as a hidden dependency.

We may need to update the README.md file later to inform users to use global unique service names.

Reply:
Ok. Let's write detail document after we implement all basic things